src/ngx_http_lua_kong_module.c

```diff
@@ -383,6 +391,22 @@ ngx_http_lua_kong_set_upstream_ssl(ngx_http_request_t *r, ngx_connection_t *c)
         goto failed;
     }
 
+skip_client_cert_key:
+
+    if (ctx->upstream_trusted_store != NULL) {
+        store = ctx->upstream_trusted_store;
+
+        if (SSL_set1_verify_cert_store(sc, store) == 0) {
+            ngx_ssl_error(NGX_LOG_ALERT, c->log, 0, "SSL_set1_verify_cert_store() failed");
+            goto failed;
+        }
+    }
+
+    if (ctx->upstream_ssl_verify_depth > 0) {
```

Comment:
This doesn't seems right. Nginx and OpenSSL both accepts `verify depth = 0` as an acceptable value, therefore we should not use `0` as the "not set flag".

The typical two ways in Nginx of handling this:

1. Add a bitfield `ngx_uint_t upstream_ssl_verify_depth;` at the end. Make a comment to indicate that it can be a 1-bit bitfield. Since no other bit-field is present, making it a single bit right now causes unnecessary `AND` operation when reading so it is indicated in the comment only. When the next bit flag got added it will be changed to a bit field. (See http://lxr.nginx.org/source/src//core/ngx_cycle.h#0119, `ngx_uint_t                transparent;  /* unsigned  transparent:1; */` definition as an example)
2. Simply use `ngx_int_t` for the depth and use `-1` as the "unset" value

Personally 1 is my favorite. It's a bit more work, but safer, as the zeroed `ctx` allocated by `pcalloc` is already in the  "unset" state. For 2 `ctx->upstream_ssl_verify_depth`must be explicitly initialized to `-1` after allocation.

Reply:
Discussed IRL and took approach 1 as path forward.